src/onnx/parse_resize.cpp

```diff
@@ -287,61 +336,102 @@ struct parse_resize : op_parser<parse_resize>
-                std::vector<size_t> in_idx(out_idx_v.size());
-                for(auto ii = 0; ii < in_lens.size(); ++ii)
-                {
-                    auto idx_val = idx_op(in_lens[ii], out_lens[ii], out_idx_v[ii], vec_scale[ii]);
-                    in_idx[ii]   = nearest_op(in_lens[ii], idx_val);
-                }
-
-                ind[out_idx] = static_cast<int64_t>(in_s.index(in_idx));
-            });
-
-            shape ind_s{shape::int32_type, out_lens};
-            auto ins_ind = info.add_literal(literal(ind_s, ind));
-            return info.add_instruction(make_op("gather", {{"axis", 0}}), rsp, ins_ind);
+            if(args[0]->get_shape().dynamic() or not is_static_scale_input)
+            {
+                // Resize's compute_shape() will read scales_sizes_arg as "scales" or "sizes"
+                // depending on its data type
+                return info.add_instruction(
+                    make_op("resize",
+                            {{"nearest_mode", nearest_mode},
+                             {"coordinate_transformation_mode", coord_trans_mode}}),
+                    args[0],
+                    scales_sizes_arg);
+            }
+            else
+            {
+                // If there are no dynamic shapes and size/scale attributes are literals, then
+                // all the indexes can be calculated now at compile time and
+                // the Resize can be accomplished with Gather operation.  Preferred for
+                // better performance.
+
+                shape out_s{in_s.type(), out_lens};
+                std::size_t out_elements = out_s.elements();
+
+                return make_gather_instruction(
+                    info, out_elements, in_s, out_s, in_lens, out_lens, vec_scale, args[0]);
+
+                // TODO: to make Onnx resize ALWAYS parse to a Resize op, replace the above line
+                //  with the following.  But it will break a lot of onnx tests.
+                //
+                //        ***  Call resize with only one argument and static scales or sizes
+                //        attribute.
+                //
+                //  Default "mode" attribute is "nearest"
+                // if(scales_sizes_arg->get_shape().type()  == shape::int64_type )
+                // {
+                //     return info.add_instruction(make_op("resize", { {"nearest_mode",
+                //     nearest_mode}
+                //         , {"coordinate_transformation_mode", coord_trans_mode}, {"sizes",
+                //         in_lens}}), args[0]);
+                // }
+                // else
+                // {
+                //     return info.add_instruction(make_op("resize", { {"nearest_mode",
+                //     nearest_mode}
+                //         , {"coordinate_transformation_mode", coord_trans_mode}, {"scales",
+                //         vec_scale}}), args[0]);
+                // }
```

Comment:
Note that this can only be done if a compiler pass is made to convert to the gather version from the Resize op. Additionally the current Resize op will need to handle all the needed cases.

Reply:
Added your remark to the code comments.